src/status_im/ui/screens/wallet_connect/session_proposal/views.cljs

```diff
@@ -66,17 +66,23 @@
                          :background-color (:color @selected-account)
                          :on-press on-press}]]))
 
-(defview success-sheet-view [{:keys [topic]}]
+(defview success-sheet-view [{:keys [wc-version] :as session-data}]
   (letsubs [visible-accounts @(re-frame/subscribe [:visible-accounts-without-watch-only])
             dapps-account [:dapps-account]
             sessions [:wallet-connect/sessions]
+            sessions-legacy [:wallet-connect-legacy/sessions]
             managed-session [:wallet-connect/session-managed]]
-    (let [{:keys [peer state] :as session} (first (filter #(= (:topic %) topic) sessions))
-          {:keys [accounts]} state
+    (let [{:keys [topic]} (when-not (= wc-version 1) session-data)
+          peerId (when (= wc-version 1) (get-in session-data [:params 0 :peerId]))
+          session-legacy (when (= wc-version 1) (first (filter #(= (get-in % [:params 0 :peerId]) peerId) sessions-legacy)))
+          {:keys [peer state] :as session} (first (filter #(= (:topic %) topic) sessions))
+          {:keys [params]} (when (= wc-version 1) session-legacy)
           {:keys [metadata]} peer
-          {:keys [name icons]} metadata
+          {:keys [peerMeta]} (first params)
+          {:keys [accounts]} (if (= wc-version 1) (first params) state)
+          {:keys [name icons]} (if (= wc-version 1) peerMeta metadata)
           icon-uri (when (and icons (> (count icons) 0)) (first icons))
```

Comment:
`(> (count icons) 0)` could be `(pos? (count icons))` not a big deal (one 0 less)